pkg/resourcecollector/endpoint.go

```diff
@@ -0,0 +1,36 @@
+package resourcecollector
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func (r *ResourceCollector) endpointsToBeCollected(
+	object runtime.Unstructured,
+) (bool, error) {
+	var endpoint v1.Endpoints
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(object.UnstructuredContent(), &endpoint); err != nil {
+		return false, fmt.Errorf("error converting to endpoint: %v", err)
+	}
+	if endpoint.Annotations != nil {
+		// collect manually created endpoint resources
+		if _, ok := endpoint.Annotations[v1.LastAppliedConfigAnnotation]; !ok {
+			return false, nil
+		}
+		// collect endpoint which has include-resources annotation applied
+		if _, ok := endpoint.Annotations[IncludeResources]; ok {
```

Comment:
Also check for skip resources?

Reply:
we do have check for ` skip resource` for all of the resources in `objectsToBeCollected`